### ISSUE TITLE
IFC support on M1, error "IfcOpenShell not built for 'darwin/64bit/python3.10'"

### ISSUE THREAD

**Issue by tjol23** [IfcOpenShell/IfcOpenShell#3980]
Hi,

Can't import IFC files on my MacbookPro arm M1 running Sonoma 14.1 (23B74), already tried:

1.  Installing ifcopenshell via pip (/Applications/FreeCAD.app/Contents/Resources/bin/python -m pip install ifcopenshell)
2. Installing the pre-compiled wheel. From [ifcOpenBot at github](https://github.com/IfcOpenBot/IfcOpenShell) followed the link to the PyPi link and downloaded the [wheel for Pyton 3.10 and arm64](https://files.pythonhosted.org/packages/21/ed/9109c91a5201742583ea6208a6a3980abe2b1c27fdac7d9db01e19bebadf/ifcopenshell-0.7.0.231018-py310-none-macosx_11_0_arm64.whl) and using pip, wheel and setuptools
3. Installing the pre-built packages (as seen on IfcOpenShell documentation [here](https://blenderbim.org/docs-python/ifcopenshell-python/installation.html))

Result is:

```
>>> import ifcopenshell
Traceback (most recent call last):
  File "/Applications/FreeCAD.app/Contents/Resources/lib/python3.10/site-packages/ifcopenshell/__init__.py", line 61, in <module>
    from . import ifcopenshell_wrapper
  File "/Applications/FreeCAD.app/Contents/Resources/lib/python3.10/site-packages/ifcopenshell/ifcopenshell_wrapper.py", line 13, in <module>
    from . import _ifcopenshell_wrapper
ImportError: dlopen(/Applications/FreeCAD.app/Contents/Resources/lib/python3.10/site-packages/ifcopenshell/_ifcopenshell_wrapper.so, 0x0002): tried: '/Applications/FreeCAD.app/Contents/Resources/lib/python3.10/site-packages/ifcopenshell/_ifcopenshell_wrapper.so' (code signature in <F3F8D36C-BDAF-3959-9A84-C6CD6B59634C> '/Applications/FreeCAD.app/Contents/Resources/lib/python3.10/site-packages/ifcopenshell/_ifcopenshell_wrapper.so' not valid for use in process: mapping process and mapped file (non-platform) have different Team IDs), '/System/Volumes/Preboot/Cryptexes/OS/Applications/FreeCAD.app/Contents/Resources/lib/python3.10/site-packages/ifcopenshell/_ifcopenshell_wrapper.so' (no such file), '/Applications/FreeCAD.app/Contents/Resources/lib/python3.10/site-packages/ifcopenshell/_ifcopenshell_wrapper.so' (code signature in <F3F8D36C-BDAF-3959-9A84-C6CD6B59634C> '/Applications/FreeCAD.app/Contents/Resources/lib/python3.10/site-packages/ifcopenshell/_ifcopenshell_wrapper.so' not valid for use in process: mapping process and mapped file (non-platform) have different Team IDs)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Applications/FreeCAD.app/Contents/Resources/lib/python3.10/site-packages/ifcopenshell/__init__.py", line 69, in <module>
    raise ImportError("IfcOpenShell not built for '%s'" % python_distribution)
ImportError: IfcOpenShell not built for 'darwin/64bit/python3.10'
>>>
```

My system (I'm using ifcopenshell from Freecad):

```
OS: macOS 14.1
Word size of FreeCAD: 64-bit
Version: 0.21.1.33694 (Git)
Build type: Release
Branch: (HEAD detached at 0.21.1)
Hash: f6708547a9bb3f71a4aaade12109f511a72c207c
Python 3.10.12, Qt 5.15.8, Coin 4.0.0, Vtk 9.2.5, OCC 7.6.3
Locale: C/Default (C)
Installed mods: 
  * FreeCAD-NativeIFC 2023.10.27
  * BIM 2021.12.0
```

Any one with ideas on how to solve it?

**Comment by aothms** [comment]
This is the "hardened runtime" https://developer.apple.com/documentation/security/hardened_runtime at work. And will likely require code signing using the same signature for freecad and ifcopenshell to work. I would try and see if you can granularly disable the hardened runtime for FreeCAD. I think you'll get this error on every compiled python extension it tries to load.

**Comment by tjol23** [comment]
Thanks for the quick response, it looks like that's the root cause.

I've been going around codesign to see how can I do it but have been unsuccessful until now. Do you have any idea on how to disable hardened runtime for an installed app?

**Comment by tjol23** [comment]
Allow me to add here the answer from Freecad devs, for present and future reference:

> I suspect that this will not work with the current signed release because of a lack of entitlements. That will be fixed in 0.21.2 in a few weeks.

Freecad forum topic [here](https://forum.freecad.org/viewtopic.php?p=717848)

**Comment by aothms** [comment]
Ok, thanks for the extra info. We're fully dependent on the community for these kind of things. We don't have anybody in the core team who is a regular macos user.
